### PR TITLE
Support multibyte character in data_format function

### LINKED ISSUE #13022: GENERIC_INTERNAL_ERROR: String index out of range when using multibyte string in date_format's format
I had an error when executing this query.

```sql
SELECT DATE_FORMAT(now() AT TIME ZONE 'Asia/Tokyo', '%Y年%m月%d日')
```

I know the solution is not using multibyte string in date_format's format, but I want to know the reason.

https://github.com/prestodb/presto/blob/28ec8f43fec01a6cd7bdec2d48852bf133ad2896/presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java#L667-L685

### PR DESCRIPTION
**Pull request by ebyhr**
Fix #13022

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctionsBase.java [hunk 1]
```diff
@@ -756,6 +756,7 @@ public void testDateFormat()
         assertFunction("date_format(" + dateTimeLiteral + ", '%g')", VARCHAR, "g");
         assertFunction("date_format(" + dateTimeLiteral + ", '%4')", VARCHAR, "4");
         assertFunction("date_format(" + dateTimeLiteral + ", '%x %v')", VARCHAR, "2001 02");
+        assertFunction("date_format(" + dateTimeLiteral + ", '%Y年%m月%d日')", VARCHAR, "2001年01月09日");
```
Comment (highker) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctionsBase.java]: We don't usually use non-ascii code in our codebase. Could you convert it into unicode encoding?
```java
'%Y\u5e74%m\u6708%d\u65e5'

Comment (ebyhr) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctionsBase.java]: Sure. Let me update.

**Comment by highker** [comment]
After I clicked "merge", I found a typo in the title ..... "data" -> "date"

**Comment by ebyhr** [comment]
Oh, sorry for my mistake...!

**Pull request merged**